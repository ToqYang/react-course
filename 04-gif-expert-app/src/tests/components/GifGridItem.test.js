import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Tests component GifGridItem", () => {
  const url =
    "https://media2.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=a652ae1db1f3uib8icf4f8d5yjlm65vkoog62yb4fh30nt2g&rid=giphy.gif";
  const title = "opm GIF";
  const wrapper = shallow(<GifGridItem url={url} title={title} />);

  test("Test if the coponent update correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener el parrafo con el titulo ", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("img igual al props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener fadeIn", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__fadeIn")).toBe(true);
  });
});
