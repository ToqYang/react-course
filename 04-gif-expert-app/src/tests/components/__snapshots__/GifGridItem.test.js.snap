// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests component GifGridItem Test if the coponent update correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="opm GIF"
    src="https://media2.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=a652ae1db1f3uib8icf4f8d5yjlm65vkoog62yb4fh30nt2g&rid=giphy.gif"
  />
  <p>
    opm GIF
  </p>
</div>
`;
